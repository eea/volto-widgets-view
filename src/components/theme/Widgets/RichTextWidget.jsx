import React from 'react';
import cx from 'classnames';

export const RichTextWidget = ({ children, className }) => (
  <div
    className={cx(className, 'richtext', 'widget')}
    dangerouslySetInnerHTML={{ __html: children.data }}
  />
);
