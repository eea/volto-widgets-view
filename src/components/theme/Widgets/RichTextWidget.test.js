import React from 'react';
import renderer from 'react-test-renderer';
import { RichTextWidget } from './RichTextWidget';

describe('RichTextWidget', () => {
  it('renders a rich text view widget component', () => {
    const component = renderer.create(
      <RichTextWidget className="metadata">
        {{ data: '<b>Foo bar</b>' }}
      </RichTextWidget>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
