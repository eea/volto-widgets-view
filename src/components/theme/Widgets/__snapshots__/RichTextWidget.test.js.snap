// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RichTextWidget renders a rich text view widget component 1`] = `
<div
  className="metadata richtext widget"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<b>Foo bar</b>",
    }
  }
/>
`;
